test/api/v3/integration/members/POST-transfer_gems.test.js

```diff
@@ -59,6 +59,48 @@ describe('POST /members/transfer-gems', () => {
     });
   });
 
+  it('returns error when to user has blocked the sender', async () => {
+    let receiverBlocks = await generateUser({'inbox.blocks': [userToSendMessage._id]});
+
+    await expect(userToSendMessage.post('/members/transfer-gems', {
+      message,
+      gemAmount,
+      toUserId: receiverBlocks._id,
+    })).to.eventually.be.rejected.and.eql({
+      code: 401,
+      error: 'NotAuthorized',
+      message: t('notAuthorizedToSendMessageToThisUser'),
+    });
+  });
+
+  it('returns error when sender has blocked to user', async () => {
+    let sender = await generateUser({'inbox.blocks': [receiver._id]});
+
+    await expect(sender.post('/members/transfer-gems', {
+      message,
+      gemAmount,
+      toUserId: receiver._id,
+    })).to.eventually.be.rejected.and.eql({
+      code: 401,
+      error: 'NotAuthorized',
+      message: t('notAuthorizedToSendMessageToThisUser'),
+    });
+  });
+
+  it('returns an error when chat privileges are revoked', async () => {
```

Comment:
If the recipient has chat permissions revoked, sending gems to them is allowed.

If the sender has site-wide chat privileges revoked, or has been personally blocked by the recipient, then they should not be able to send gems.

Does that cover what you're asking and/or what the code is doing / not doing?  (I haven't checked the code in detail, sorry. :( )